### DISCUSSION TITLE
Migrate int id to uuid string using prisma

### DISCUSSION THREAD

**Discussion by tkow** [prisma/prisma#13827] [Q&A] [answered]
I want to my database column int id to string uuid format.

I rewrote prisma code

```
  id                   Int       @id @default(autoincrement())
```
to
```
  id                   String               @id @default(uuid())
```

Above, migration works but, the id just was stringified sequence like '1' or '2', and I want to reset id to uuid. But I didn't find this uuid function which algorithm derived from. What the best way to reassign id? I can generate uuid from node clients? If so,

```
await client.table.updateMany({data: {id: uuid()}})
```

is suitable?

I found explanation in [referred to document](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#uuid)  say this is derived from prisma.

```
 You can still use uuid() when using introspection by manually 
 changing your Prisma schema and generating Prisma Client,
 in that case the values will be generated by Prisma's query engine.
```

but can we call it from client api?

**Comment by tkow** [comment, marked as answer]
I found prisma using UUID version4 as default, so I can use  uuid v4 library to generate whatever using programming languages.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
